src/shared/services/payment-info.service.ts

```diff
@@ -36,6 +30,14 @@ export class PaymentInfoService {
     if (jwt && !jwt.blockchains.includes(dto.asset.blockchain))
       throw new BadRequestException('Asset blockchain mismatch');
 
+    if ('paymentMethod' in dto && dto.paymentMethod === FiatPaymentMethod.CARD) {
+      if (!dto.currency.cardSellable) throw new BadRequestException('Currency not sellable per Card');
+      if (!dto.asset.cardSellable) throw new BadRequestException('Asset not buyable per Card');
+    } else if ('paymentMethod' in dto && dto.paymentMethod === FiatPaymentMethod.INSTANT) {
+      if (!dto.currency.instantSellable) throw new BadRequestException('Currency not sellable per Instant');
+      if (!dto.asset.instantSellable) throw new BadRequestException('Asset not buyable per Instant');
+    }
```

Comment:
There might be no currency in the DTO here?

Reply:
Nope not possible in this case I think. Currency IsNotEmptyObject in dto's with paymentMethod 👍